addons/mail/models/mail_message.py

```diff
@@ -825,56 +808,78 @@ def unstar_all(self):
         """ Unstar messages for the current partner. """
         partner = self.env.user.partner_id
         starred_messages = self.search([('starred_partner_ids', 'in', partner.id)])
-        partner.starred_message_ids -= starred_messages
-        self.env.user._bus_send(
-            "mail.message/toggle_star", {"message_ids": starred_messages.ids, "starred": False}
-        )
+        starred_messages.toggle_message_starred()
 
     def toggle_message_starred(self):
```

Comment:
For future, it is better to have star_message and unstar_message instead of a toggle function because toggle is not idempotent.

Reply:
I agree ... wanted to remove completely toggle message, will see if I have time :)